adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -862,10 +864,10 @@ def unsubscribe(  # noqa: PLR0912, Too many branches
                             self.on_unsubscribe(self, self.user_data, t, self._pid)
                         self._subscribed_topics.remove(t)
                     return
-
-                raise MMQTTException(
-                    f"invalid message received as response to UNSUBSCRIBE: {hex(op)}"
-                )
+                if op != MQTT_PUBLISH:
+                    raise MMQTTException(
```

Comment:
you might want to add comment similar to the one in `subscribe()`, explain the rationale.